File: patterns/switch-dimmer/rules-js/switch-dimmer-light.js
let myColorLightCommand = "INCREASE";

rules.JSRule({
  name: "MyColorLight_OnOff",
  triggers: [triggers.ItemCommandTrigger("MySwitch1UpShortPress", "ON")],
  execute: (event) => {
    const brightness = items.getItem("MyColorLight").state; // HSB-Helligkeit
    if (parseFloat(brightness.split(",")[2]) === 0) {
      myColorLightCommand = "INCREASE";
      items.getItem("MyColorLight").sendCommand("ON");
    } else {
      myColorLightCommand = "DECREASE";
      items.getItem("MyColorLight").sendCommand("OFF");
    }
  }
});

rules.JSRule({
  name: "MyColorLight_Dimmer",
  triggers: [triggers.ItemCommandTrigger("MySwitch1UpLongPress", "ON")],
  execute: (event) => {
    items.getItem("MyColorLight").sendCommand(myColorLightCommand);
  }
});

File: patterns/switch-dimmer/rules-python/switch_dimmer_light.py
from openhab import rule, Registry
from openhab.triggers import ItemCommandTrigger

my_color_light_command = "INCREASE"


@rule(triggers=[ItemCommandTrigger("MySwitch1UpShortPress", "ON")])
class MyColorLightOnOff:
    def execute(self, module, input):
        global my_color_light_command
        hsb = str(Registry.getItem("MyColorLight").state)
        brightness = float(hsb.split(",")[2])
        if brightness == 0:
            my_color_light_command = "INCREASE"
            Registry.getItem("MyColorLight").sendCommand("ON")
        else:
            my_color_light_command = "DECREASE"
            Registry.getItem("MyColorLight").sendCommand("OFF")


@rule(triggers=[ItemCommandTrigger("MySwitch1UpLongPress", "ON")])
class MyColorLightDimmer:
    def execute(self, module, input):
        Registry.getItem("MyColorLight").sendCommand(my_color_light_command)
